packages/lightning-lsp-common/src/resources/sfdx/typings/lds.d.ts

```diff
@@ -547,3 +547,814 @@ declare module 'lightning/uiRecordApi' {
+        frequency: 'weekly' | 'Weekly';
+        /** Days of the week on which the schedule will run. */
+        daysOfWeek: Array<string>;
+    }
+
+    /**
+     * Weekly schedule on which to execute some type of job.
+     *
+     * https://developer.salesforce.com/docs/atlas.en-us.bi_dev_guide_rest.meta/bi_dev_guide_rest/bi_resources_appendix.htm#WeeklyScheduleRepresentation
+     *
+     * Keys:
+     *    id (string): assetId
+     */
+    export interface WeeklyScheduleRepresentation extends ScheduleRepresentation {
+        frequency: 'weekly' | 'Weekly';
+        /** Days of the week on which the schedule will run. */
+        daysOfWeek: Array<string>;
+    }
+
+    /**
+     * Creates a Tableau CRM dataflow job, which is the equivalent of clicking Run Now for a data prep recipe, a data sync,
+     * or a dataflow in the Tableau CRM Data Manager UI.
+     *
+     * @param dataflowJob.dataflowId The dataflow, data prep recipe, or data sync ID for the job.
+     * @param dataflowJob.command The job command to execute. Must be `Start` to create a dataflow job.
+     * @return A promise that will resolve to the dataflow job response.
+     */
+    export function createDataflowJob({ dataflowJob }: { dataflowJob: DataflowJobInputRepresentation }): Promise<DataflowJobRepresentation>;
+
+    /**
+     * Deletes a specific Tableau CRM dataset by ID or developer name.
+     *
+     * @param datasetIdOrApiName The ID or developer name of the dataset.
+     * @return A promise that will resolve on completion.
+     */
+    export function deleteDataset({ datasetIdOrApiName }: { datasetIdOrApiName: string }): Promise<void>;
+
+    /**
+     * Deletes a specific Tableau CRM data prep recipe by ID.
+     *
+     * @param id The ID of the data prep recipe.
+     * @return A promise that will resolve on completion.
+     */
+    export function deleteRecipe({ id }: { id: string }): Promise<void>;
+
+    /**
+     * Wire adapter to execute a Tableau CRM query.
+     *
+     * @param query.query The query string to execute.
+     * @param query.queryLanguage The query language. Valid values are `SAQL` or `SQL`. The default is `SAQL`.
+     * @param query.timezone The timezone for the query.
+     */
+    export function executeQuery(query: SaqlQueryInputRepresentation): void;
+
+    /**
+     * Wire adapter to retrieve the Analytics limits for Tableau CRM.
+     *
+     * @param licenseType The response includes dataflow jobs with this license type. Valid values are `EinsteinAnalytics` or `Sonic`.
+     * @param types The types of limits used in Tableau CRM.
+     *              Valid values are `BatchTransformationHours`, `DatasetQueries`, `DatasetRowCount`,
+     *              `MaxDailyRowsHighOutputCon`, `MaxDailyRowsLowOutputCon`, `MaxDailyRowsMedOutputCon`,
+     *              `MaxDailySizeHighOutputCon`, `MaxDailySizeLowOutputCon`, `MaxDailySizeMedOutputCon`,
+     *              or `OutputLocalConnectorVolume`.
```

Comment:
If you'd like, you could have type checking for the arguments using a type alias:

```typescript
type LicenseType = 'EinsteinAnalytics' | 'Sonic';
type LimitType = 'BatchTransformationHours' | 'DatasetQueries' | 'DatasetRowCount' // ...
export function getAnalyticsLimits(licenseType: LicenseType, types?: LimitType[]): void;
```

Reply:
So I didn't do that originally for a few reasons:
1. I saw that another enum-ish field (`FieldRepresentation.extraType`) just did `string`.
2. This function can only be used with `@wire`, and therefore is allowed to be bound ala `@wire(getAnalyticsLimits, { licenseType: '$someVariable' })`, which, in type terms, is just a `string`.
3. The typedef for `@wire` currently has the 2nd arg (which corresponds to an object structure of the args for the 1st arg method) as `any`, so you don't get any completions or type checks there anyways.

I think I'm going to leave this as-is for now, until if/when there's better intellisense/type-checking around the 2nd arg of `@wire`.

Reply:
Makes sense, thanks for the additional context @smithgp